app/src/ui/stashing/stash-diff-viewer.tsx

```diff
@@ -93,22 +107,43 @@ const Header: React.SFC<{
   stashEntry: IStashEntry
   repository: Repository
   dispatcher: Dispatcher
+  isWorkingTreeClean: boolean
 }> = props => {
+  const { dispatcher, repository, stashEntry, isWorkingTreeClean } = props
+
   const onClearClick = () => {
-    props.dispatcher.dropStash(props.repository, props.stashEntry)
+    dispatcher.dropStash(repository, stashEntry)
   }
+
   const onSubmitClick = () => {
-    props.dispatcher.popStash(props.repository, props.stashEntry)
+    dispatcher.popStash(repository, stashEntry)
   }
+
+  const restoreMessage = isWorkingTreeClean ? (
+    'This will pop this stash as changed files.'
```

Comment:
```suggestion
    'Restore will move your stashed files to the Changes list.'
```

Reply:
This is based on @tierninho's suggestion after discussing it in our weekly meeting.